test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,439 @@
+        .then(function(accessToken) {
+          assertGoodToken(accessToken, userFromOneModel);
+        });
+    });
+
+    it('works for a second user model and valid credentials', function() {
+      return AnotherUser.login(commonCredentials)
+        .then(function(accessToken) {
+          assertGoodToken(accessToken, userFromAnotherModel);
+        });
+    });
+
+    it('fails when credentials are not correct', function() {
+      return OneUser.login({email: 'foo@bar.com', password: 'invalid'})
+        .then(
+          function onSuccess() {
+            throw new Error('OneUser.login() should have failed');
+          },
+          function onError(err) {
+            expect(err).to.have.property('code', 'LOGIN_FAILED');
+          }
+        );
+    });
+  });
+
+  function assertGoodToken(accessToken, user) {
+    if (accessToken instanceof AccessToken) {
+      accessToken = accessToken.toJSON();
+    }
+    expect(accessToken.id, 'token id').to.have.lengthOf(64);
+    expect(accessToken).to.have.property('userId', user.id);
+    expect(accessToken).to.have.property('principalType', user.constructor.definition.name);
+  }
+
+  describe('User.logout', function() {
+    it('should logout a user from user model 1 without logging out user from model 2',
+    function() {
+      var tokenOfOneUser;
+      return Promise.all([
+        OneUser.login(commonCredentials),
+        AnotherUser.login(commonCredentials),
+      ])
+      .spread(function(t1, t2) {
+        tokenOfOneUser = t1;
+        return OneUser.logout(tokenOfOneUser.id);
+      })
+      .then(function() {
+        return AccessToken.find({});
+      })
+      .then(function(allTokens) {
+        var data = allTokens.map(function(token) {
+          return {userId: token.userId, principalType: token.principalType};
+        });
+        expect(data).to.eql([
+          // no token for userFromAnotherModel
+          {userId: userFromAnotherModel.id, principalType: 'AnotherUser'},
+        ]);
+      });
+    });
+  });
+
+  describe('Password Reset', function() {
+    describe('User.resetPassword(options)', function() {
+      var options = {
+        email: 'foo@bar.com',
+        redirect: 'http://foobar.com/reset-password',
+      };
+
+      it('should create a temp accessToken to allow a user to change password', function() {
+        return Promise.all([
+          OneUser.resetPassword({email: options.email}),
+          waitForResetRequestAndVerify,
+        ]);
+      });
+
+      function waitForResetRequestAndVerify() {
+        return waitForEvent(OneUser, 'resetPasswordRequest')
+        .then(function(info) {
+          assertGoodToken(info.accessToken, userFromOneModel);
+          return info.accessToken.user.getAsync();
+        })
+        .then(function(user) {
+          expect(user).to.have.property('id', userFromOneModel.id);
+          expect(user).to.have.property('email', userFromOneModel.email);
+        });
+      }
+    });
```

Comment:
isolated the whole logic of verifying the temp accessToken info in a dedicated helper, as per your suggestion